scalableminds/webknossos
# Reactify Task Creation

## ATTENTION: DO NOT MERGE BEFORE BRAVE NEW WORLD RELEASE.

I reactified the Task Creation view. Forms are similar to project, script creation etc. Please see the mailable description below for detailed changes.

### Mailable description of changes:
- [Admin.Task Creation] Updated the task creation form
- [Admin.Task Creation] Added support for live form input validation
- [Admin.Task Creation] Joined manual task creation and NML uploading into a single tab since they handled almost identical
- [Admin.Task Creation] Updated bulk task creation with live validation feedback, including the line numbers
- [Admin.Task Creation] Added support for the use of CSV files for bulk task creation
- [Admin.Task Creation] Added progress bar for bulk creation. Tasks are submitted to the server and created in batches of 100 items.

### Steps to test:
- Create new Task using the form
- Try to enter invalid values. Form validation should warn you and prevent you from submitting.
- Create a task by uploading an NML or zip of NMLs
- Bulk created some tasks. Try to break task parser.
- Bulk create some tasks from CSV files. Again, try CSV with invalid values. 
- Double check the network tab. Bulk creation should be split into batches of 100 tasks. 
- Try to edit an existing task.

### Issues:
- part of #2009

This PR will most likely need some serious merging with the brave new world changes.

------
